### PR TITLE
[easee] Migrate charger state API endpoint, add channels

### PR DESCRIPTION
**Pull request by alexf2015**
This PR implements some points mentioned in issue #14037

1. the charger state API endpoint is now migrated to the new API
2. some missing channels were added
3. changed behaviour in error case: instead of taking the bridge offline only the dedicated thing will be put offline. Thus a faulty charger config will not take down the whole bridge including all things.

### CODE REVIEW

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks! I have provided only some minor suggestions. Also, please add update instructions for the channel added - see https://next.openhab.org/docs/developer/bindings/thing-xml.html#updating-thing-types

File: bundles/org.openhab.binding.easee/src/main/resources/OH-INF/thing/charger-channel-groups.xml [hunk 1]
```diff
@@ -26,6 +26,18 @@
 				<label>Session Energy</label>
 				<description>Energy for current session.</description>
 			</channel>
+			<channel id="energyPerHour" typeId="type-energy">
+				<label>Energy per Hour</label>
+				<description>Energy transferred per hour.</description>
+			</channel>
+			<channel id="wiFiRSSI" typeId="type-integer">
```
[the remark this answers is not in the document]
Comment (alexf2015) [reply bundles/org.openhab.binding.easee/src/main/resources/OH-INF/thing/charger-channel-groups.xml:33]: I am not sure if mW is the correct unit. Usually it is measured in dBm. Transformation to mW is not easy and in addition I am in doubt that people will understand the transformed values. people will now that -30dBm is a really good signal and -60dBm is still good. On the other hand values like 1 µW or 1 nW are not easy to understand.

[the remark this answers is not in the document]
Comment (jlaur) [reply bundles/org.openhab.binding.easee/src/main/resources/OH-INF/thing/charger-channel-groups.xml:33]: I agree dBm is the unit normally used for measuring RSSI, so hence my suggestion to apply it. See for example #14319.

[the remark this answers is not in the document]
Comment (alexf2015) [reply bundles/org.openhab.binding.easee/src/main/resources/OH-INF/thing/charger-channel-groups.xml:33]: ok now I see that I overlooked that you suggested ```Units.DECIBEL_MILLIWATTS``` and not ```Units.MILLIWATTS``` which most likely does not exist at all...
I will change it accordingly.

File: bundles/org.openhab.binding.easee/src/main/resources/OH-INF/thing/easee-readonly-channel-types.xml [hunk 2]
```diff
@@ -3,6 +3,12 @@
 	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
 	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
 	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+	<channel-type id="type-rssi">
+		<item-type>Number:Power</item-type>
+		<label>Received Signal Strength Indicator</label>
```
Comment (jlaur) [inline bundles/org.openhab.binding.easee/src/main/resources/OH-INF/thing/easee-readonly-channel-types.xml:8]:
```suggestion
		<label>Received Signal Strength Indicator</label>
		<category>QualityOfService</category>
```

**Comment by jlaur** [comment]
> Also, please add update instructions for the channel added - see https://next.openhab.org/docs/developer/bindings/thing-xml.html#updating-thing-types

Just in case you missed this - please provide also the update instructions.

**Comment by alexf2015** [comment]
Sure I will add migration. In general this is a really good idea. Unfortunately I discovered some issues in my other PR.
1. I am using Eclipse and there are some issues that prevent the whole thing from starting up (in Eclipse) as soon as there are migration statements present.
2. This should not affect this binding as the changes do not affect any channels that have properties but channel properties are not yet supported.

**Comment by alexf2015** [comment]
tested migration successfully.
```thingTypeVersion``` is now set to ```2``` as there is another pull request pending which already uses version 1. Depending on which is merged back earlier I need to adjust.

**Comment by Diapolo** [comment]
Easee is changing their API endpoint to `api.easee.com` on September 1st 2023 (see here https://developer.easee.cloud/docs). @alexf2015 Perhaps you could integrate that change here or create a new pull for that? And any idea when this pull here will be merged? I'd love to kick out my own rule based query/parsing stuff and just use this nice addon :). Thanks for your work!

**Comment by Diapolo** [comment]
What can be done to get this finally in? I'm waiting for this.

**Comment by jlaur** [comment]
> tested migration successfully.
> `thingTypeVersion` is now set to `2` as there is another pull request pending which already uses version 1. Depending on which is merged back earlier I need to adjust.

Sorry for the long delay, I must have somehow missed it, since it looks quite up-to-date already with latest comments. Can you rebase towards main now that #15229 was merged?

**Comment by alexf2015** [comment]
> I mixed the two PR's - #14866 is now merged, so you can rebase and change `thingTypeVersion` to 1.
done

**Review by jlaur — APPROVED** [review]
Thank you for these improvements!

**Comment by Diapolo** [comment]
@alexf2015 AFAIK there are still some values missing, could you take a look at:
localRSSI
outputPhase
chargerRAT
circuitTotalAllocatedPhaseConductorCurrentL1
circuitTotalAllocatedPhaseConductorCurrentL2
circuitTotalAllocatedPhaseConductorCurrentL3
circuitTotalPhaseConductorCurrentL1
circuitTotalPhaseConductorCurrentL2
circuitTotalPhaseConductorCurrentL3
wiFiAPEnabled
offlineMaxCircuitCurrentP1
offlineMaxCircuitCurrentP2
offlineMaxCircuitCurrentP3
eqAvailableCurrentP1
eqAvailableCurrentP2
eqAvailableCurrentP3
deratedCurrent
deratingActive
errors

I really don't know if these are used anymore, but they are still in the JSON reply from the deprecated endpoint (which I now replaced to just your addon, so I don't get them anymore).

**Pull request merged**